repose-aggregator/docs/build.gradle

```diff
@@ -0,0 +1,286 @@
+                dir: workingDir,
+            )
+        }
+
+        file("${project.buildDir}/gh-pages/versions/$version").deleteDir()
+        copy {
+            from("${project.projectDir}/src/resources")
+            into("${project.buildDir}/gh-pages/")
+        }
+        copy {
+            from "${project.buildDir}/asciibinder/_package/main/"
+            into "${project.buildDir}/gh-pages/versions/"
+        }
+
+        // Setup the version utilities
+        def versionsDir = file("${project.buildDir}/gh-pages/versions")
+        def versionComparator = new Comparator() {
+            @Override
+            int compare(Object o1, Object o2) {
+                List<String> v1Parts = ((String) o1).tokenize('.')
+                List<String> v2Parts = ((String) o2).tokenize('.')
+
+                int minLen = Math.min(v1Parts.size(), v2Parts.size())
+
+                // Compare each number from the versions until either the numbers do not match,
+                // or the end of a version string is hit.
+                for (int i = 0; i < minLen; ++i) {
+                    int n1 = v1Parts[i] as int
+                    int n2 = v2Parts[i] as int
+
+                    if (n1 != n2) {
+                        return n1 <=> n2
+                    }
+                }
+
+                // If we got this far, then one version is a substring of the other.
+                // Therefore, whichever version is longer is the greater version.
+                return v1Parts.size() <=> v2Parts.size()
+            }
+        }
+
+        // Write the searchable versions list as JSON
+        def jsonFile = file("${project.buildDir}/gh-pages/versions/versions.json")
+        Files.deleteIfExists(jsonFile.toPath())
+        def jsonWriter = new PrintWriter(jsonFile)
+        jsonWriter.print("""{"versions": [""")
+        def jsonVersions = versionsDir.list(new FilenameFilter() {
+            def pattern = Pattern.compile("""^data_\\d+\\.\\d+\\.\\d+\\.\\d+\\.json\$""")
+
+            @Override
+            boolean accept(File dir, String name) {
+                return pattern.matcher(new File(name).getName()).matches()
+            }
+        }).collect { it.replace("data_", "").replace(".json", "") }
+        jsonVersions.sort(versionComparator)
+        jsonVersions.reverse(true)
+        jsonVersions.eachWithIndex { String entry, int i ->
+            if (i != 0) {
+                jsonWriter.print(',')
+            }
+            jsonWriter.print("\"${entry}\"")
+        }
+        jsonWriter.print("""]}""")
+        jsonWriter.close()
+
+        // Write the versions includes file
+        def htmlFile = file("${project.buildDir}/gh-pages/_includes/versions.html")
+        Files.deleteIfExists(htmlFile.toPath())
+        def htmlWriter = new PrintWriter(htmlFile)
+        def htmlVersions = versionsDir.list(new FilenameFilter() {
+            def pattern = Pattern.compile("""\\d+\\.\\d+\\.\\d+\\.\\d+\$""")
```

Comment:
This was originally due to their being two-sources of truth. The JSON for the searches being generated from the search data and the HTML based on the available directories. After further discussions, it was accepted that this can be a single source of truth since the docs can not be published if the search data file fails.